definitions/variable/socio-economic/carbon-price.yaml

```diff
@@ -0,0 +1,26 @@
+- Price|Carbon:
+    description: Price of carbon (as reported by the model)
+    unit: USD_2010/t CO2
+    region-aggregation:
+      - Price|Carbon (mean):
+          method: mean
+      - Price|Carbon (weighted by Emissions|CO2):
+          weight: Emissions|CO2
+          drop_negative_weights: false
+      - Price|Carbon (weighted by Final Energy):
+          weight: Final Energy
+    check-aggregate: false
+- Price|Carbon (mean):
+    description: Price of carbon (computed as unweighted mean across regions)
+    unit: USD_2010/t CO2
+    skip-region-aggregation: true
+- Price|Carbon (weighted by Emissions|CO2):
+    description: Price of carbon (weighted by regional CO2 emissions)
+    unit: USD_2010/t CO2
+    skip-region-aggregation: true
+    note: Regional CO2 emissions can turn negative at different points in time, which
```

Comment:
I wonder whether aggregating by `Gross Emissions|CO2` would be preferable, avoiding this really counter-intuitive and confusing issue of negative or undefined (division by 0) carbon prices

Reply:
Not sure about that, this solution would also deviate from the "standard" solution where the price is weighted by net emissions.

Reply:
There is no "perfect" solution. In my view, the version "weighted by final energy" would be the best default setting, as it represents the angle "how large is the energy system for which a carbon price is relevant" (this could also be "primary", but I think final is a bit better), but if one takes a different angle, other aggregations can be more fitting. 

If there is a specific angle for which weighting by "gross emissions" would be better, it could be added as well, right?

Reply:
Gross and Net emissions are both fine.
Having negative mean pricing should be rare: in my understanding, this would happened if regions without NET have higher carbon prices than in regions with NET, which is really counter-intuitive and is more a modelling issue.

In the past Gross emissions were used (previous papers and the IPCC) to compute the net present value of the global carbon price in order to cover negative prices, but again I think at that time the issue was the use of a discount rate not compatible with some model results, rather than using "Gross" over "Net" emissions.

Anyway those should be moved to the "macro" list now.

Reply:
Thank for this numerical example @robertpietzcker 
I agree then.

With the caveat: carbon price trigger energy efficiency, which in this case decrease the weight applied.